### ISSUE TITLE
Fix S6966 FP: Do not use AddAsync with EF Core

### ISSUE THREAD

**Issue by grosch-intl** [SonarSource/sonar-dotnet#9301]
### Description

S6966 is being flagged on the `Add` method for `DbSet<T>` in Entity Framework.  That should not be flagged as the aync version is not supposed to be used.  Copying right from the documentation for `AddAsync`:

> This method is async only to allow special value generators, such as the one used by Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used.

### Repro steps

```csharp
using var context = new MyEfContext();
context.MyTable.Add(....)
```

### Expected behavior

Not get flagged.

### Actual behavior

Tells me to await AddAsync instead.

### Known workarounds

I could disable the rule, but I want it to tell me for everything else.  Wrapping every add in a pragma to ignore it is a bad idea.

### Related information

* Visual Studio version 2022
* MSBuild / dotnet version 8.0.204
* SonarAnalyzer.CSharp version 9.25.0.90414

**Comment by grosch-intl** [comment]
The same applies to `AddRangeAsync`.

**Comment by cremor** [comment]
Duplicate of #9269

**Comment by costin-zaharia-sonarsource** [comment]
Closing this issue as a duplicate of https://github.com/SonarSource/sonar-dotnet/issues/9269
